File: src/main/java/io/github/tanguygab/tabadditions/shared/commands/ActionBarCmd.java
package io.github.tanguygab.tabadditions.shared.commands;

import io.github.tanguygab.tabadditions.shared.TABAdditions;
import io.github.tanguygab.tabadditions.shared.features.actionbar.ActionBarManager;
import me.neznamy.tab.shared.TAB;
import me.neznamy.tab.shared.command.SubCommand;
import me.neznamy.tab.shared.platform.TabPlayer;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public class ActionBarCmd extends SubCommand {

    private final ActionBarManager manager;

    public ActionBarCmd(ActionBarManager manager) {
        super("actionbar",null);
        this.manager = manager;
    }

    @Override
    public void execute(TabPlayer sender, String[] args) {
        TABAdditions instance = TABAdditions.getInstance();

        if (args.length < 1) {
            sendMessage(sender,"&cYou have to provide a player!");
            return;
        }
        if (args.length < 2) {
            sendMessage(sender,"&cYou have to provide an actionbar!");
            return;
        }

        String name = args[0];
        String actionbar = args[1];

        if (name.equals("*")) {
            for (TabPlayer player : TAB.getInstance().getOnlinePlayers())
                manager.announceBar(player,actionbar);
            return;
        }

        boolean self = name.equalsIgnoreCase("me");
        TabPlayer player = self ? sender : instance.getPlayer(name);
        if (player == null) {
            sendMessage(sender,self ? getMessages().getCommandOnlyFromGame() : getMessages().getPlayerNotFound(name));
            return;
        }
        manager.announceBar(player,actionbar);
    }

    @Override
    public @NotNull List<String> complete(@Nullable TabPlayer sender, @NotNull String[] arguments) {
        if (arguments.length == 1) {
            List<String> players = getOnlinePlayers(arguments[0]);
            players.add("*");
            players.add("me");
            return players;
        }
        return getStartingArgument(manager.getActionBars().keySet(),arguments[1]);
    }

}

File: src/main/kotlin/io/github/tanguygab/tabadditions/shared/commands/ActionBarCmd.kt
package io.github.tanguygab.tabadditions.shared.commands

import io.github.tanguygab.tabadditions.shared.TABAdditions
import io.github.tanguygab.tabadditions.shared.features.actionbar.ActionBarManager
import me.neznamy.tab.shared.TAB
import me.neznamy.tab.shared.command.SubCommand
import me.neznamy.tab.shared.platform.TabPlayer

class ActionBarCmd(private val manager: ActionBarManager) : SubCommand("actionbar", null) {
    override fun execute(sender: TabPlayer?, args: Array<String>) {
        val instance = TABAdditions.INSTANCE

        if (args.isEmpty()) {
            sendMessage(sender, "&cYou have to provide a player!")
            return
        }
        if (args.size < 2) {
            sendMessage(sender, "&cYou have to provide an actionbar!")
            return
        }

        val name = args[0]
        val actionbar = args[1]

        if (name == "*") {
            TAB.getInstance().onlinePlayers.forEach {
                manager.announceBar(it, actionbar)
            }
            return
        }

        val self = name.equals("me", ignoreCase = true)
        val player = if (self) sender else instance.getPlayer(name)
        if (player == null) {
            sendMessage(sender, if (self) messages.commandOnlyFromGame else messages.getPlayerNotFound(name))
            return
        }
        manager.announceBar(player, actionbar)
    }

    override fun complete(sender: TabPlayer?, arguments: Array<String>) = if (arguments.size == 1)
        getOnlinePlayers(arguments[0]).apply {
            add("*")
            add("me")
        }
    else getStartingArgument(manager.actionBars.keys, arguments[1])
}
